src/main/java/org/mule/module/apikit/validation/body/form/transformation/MultipartEntityBuilder.java

```diff
@@ -65,4 +94,51 @@ protected Map<String, String> getHeaders(String headers) {
     }
     return map;
   }
+
+  public void handlePart(MultipartStream multipartStream, Parameter parameter, String name, String contentType, String fileName,
+                         String headers)
+      throws InvalidFormParameterException {
+    if (parameter != null && parameter.getFileProperties().isPresent()) {
```

Comment:
can you add here the contentType check ? to avoid handling as text binary payloads, it is another ticket, but seems to be easy to be included here W-11669811

Reply:
Will address this in a different PR for the parser wrapper

Reply:
Please review the fix as part of this [PR](https://github.com/mulesoft/apikit-rest-parser-wrapper/pull/347)